File: VB/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.Data.Filtering
Imports DevExpress.XtraPrinting

Namespace WindowsFormsApplication1
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub xrPivotGrid1_PrintFieldValue(ByVal sender As Object, ByVal e As DevExpress.XtraReports.UI.PivotGrid.CustomExportFieldValueEventArgs) Handles xrPivotGrid1.PrintFieldValue
            If e.Field IsNot Nothing AndAlso e.Field.Area = DevExpress.XtraPivotGrid.PivotArea.ColumnArea Then
                    Dim tb As TextBrick = TryCast(e.Brick, TextBrick)
                    Dim lb As LabelBrick = TryCast(tb.PrintingSystem.CreateBrick("LabelBrick"), LabelBrick)
                    lb.Padding = New PaddingInfo(2, 2, 5, 2, GraphicsUnit.Pixel)
                    lb.Angle = 90
                    lb.Text = e.Text
                    lb.Rect = GraphicsUnitConverter.DocToPixel(e.Brick.Rect)
                    tb.IsVisible = False
                    tb.PrintingSystem.Graph.DrawBrick(lb)
            End If
        End Sub

    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.Data.Filtering;
using DevExpress.XtraPrinting;

namespace WindowsFormsApplication1
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport1()
        {
            InitializeComponent();
        }

        private void xrPivotGrid1_PrintFieldValue(object sender, DevExpress.XtraReports.UI.PivotGrid.CustomExportFieldValueEventArgs e)
        {
            if (e.Field != null && e.Field.Area == DevExpress.XtraPivotGrid.PivotArea.ColumnArea)
                {
                    TextBrick tb = e.Brick as TextBrick;
                    LabelBrick lb = tb.PrintingSystem.CreateBrick("LabelBrick") as LabelBrick;
                    lb.Padding = new PaddingInfo(2, 2, 5, 2, GraphicsUnit.Pixel);
                    lb.Angle = 90;
                    lb.Text = e.Text;
                    lb.Rect = GraphicsUnitConverter.DocToPixel(e.Brick.Rect);
                    tb.IsVisible = false;
                    tb.PrintingSystem.Graph.DrawBrick(lb);
                }
        }

    }
}
